src/controller/AutoCommissioner.cpp

```diff
@@ -502,9 +511,43 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
             SetDAC(report.Get<RequestedCertificate>().certificate);
             break;
         case CommissioningStage::kSendAttestationRequest:
-            // These don't need to be deep copied to local memory because they are used in this one step then never again.
-            mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
-                .SetAttestationSignature(report.Get<AttestationResponse>().signature);
+            if (report.Get<AttestationResponse>().attestationElements.size() > kAttestationElementsLength)
+            {
+                ChipLogError(Controller, "AutoCommissioner attestationElements buffer size %d larger than cache size %d",
+                             static_cast<int>(report.Get<AttestationResponse>().attestationElements.size()),
+                             static_cast<int>(kAttestationElementsLength));
+                mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements);
+            }
+            else
+            {
+                memcpy(mAttestationElements, report.Get<AttestationResponse>().attestationElements.data(),
```

Comment:
Why not simply use a vector altogether for the mAttestationElements? It keeps its size and is trivially copyable/assignable